### PR TITLE
[TECH-509] Upgrade typescript --> v5

### PR DESCRIPTION
**Pull request by Denys-Vandebron**
branch: feature/TECH-509-upgrading-typescript

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,16 +22,16 @@
...
-    "@typescript-eslint/parser": "5.36.1",
-    "eslint": "6.5.1 || 8.23.0",
-    "eslint-config-prettier": "8.5.0",
-    "eslint-plugin-import": "2.26.0",
-    "eslint-plugin-prettier": "4.2.1",
-    "eslint-plugin-react": "7.31.6",
-    "eslint-plugin-react-hooks": "4.6.0",
-    "prettier": "2.7.1",
-    "typescript": "3.0.0 || 4.8.2"
+    "@typescript-eslint/eslint-plugin": "^6.7.2",
```
Comment (Jorg88) [inline package.json:25]: I'm not sure if this changed, but the front-end guild decided a while ago that as a best-practice we won't use `^`

Comment (Denys-Vandebron) [reply package.json:25]: Alright, no problem, I did not know this 😄  Will change

**Comment by Jorg88** [comment]
Just to be sure: you know about the release cycle (as described in the readme), which provides you with a new version that you can then use in the monorepo (pr)?

**Comment by Denys-Vandebron** [comment]
> Just to be sure: you know about the release cycle (as described in the readme), which provides you with a new version that you can then use in the monorepo (pr)?

Do you mean publishing an npm-module to the registry? Then yes :-)
I'm opting at major upgrade here (4.0.0), do you agree? @Jorg88 @lauraoneasca

**Comment by Jorg88** [comment]
Fine by me, but you can check in the front-end guild channel what others think!

**Review by lauraoneasca — APPROVED** [review]

**Pull request merged**